packages/components/src/input-control/state.js

```diff
@@ -0,0 +1,246 @@
+			return { ...state, ...fn( ...args ) };
+		}, {} );
+	};
+};
+
+/**
+ * Creates a reducer that opens the channel for external state subscription
+ * and modification.
+ *
+ * This technique uses the "stateReducer" design pattern:
+ * https://kentcdodds.com/blog/the-state-reducer-pattern/
+ *
+ * @param {Function} combinedStateReducers A custom reducer that can subscribe and modify state.
+ * @return {Function} The reducer.
+ */
+function inputControlStateReducer( combinedStateReducers ) {
+	return ( state, action ) => {
+		const nextState = { ...state };
+		const { type, payload } = action;
+
+		switch ( type ) {
+			/**
+			 * Keyboard events
+			 */
+			case actionTypes.PRESS_UP:
+				nextState.isDirty = false;
+				break;
+
+			case actionTypes.PRESS_DOWN:
+				nextState.isDirty = false;
+				break;
+
+			/**
+			 * Drag events
+			 */
+			case actionTypes.DRAG_START:
+				nextState.isDragging = true;
+				break;
+
+			case actionTypes.DRAG_END:
+				nextState.isDragging = false;
+				break;
+
+			/**
+			 * Input events
+			 */
+			case actionTypes.CHANGE:
+				nextState.error = null;
+				nextState.value = payload.value;
+
+				if ( state.isPressEnterToChange ) {
+					nextState.isDirty = true;
+				}
+
+				break;
+
+			case actionTypes.COMMIT:
+				nextState.value = payload.value;
+				nextState.isDirty = false;
+				break;
+
+			case actionTypes.RESET:
+				nextState.error = null;
+				nextState.isDirty = false;
+				nextState.value = payload.value || state.initialValue;
+				break;
+
+			case actionTypes.UPDATE:
+				if ( payload.value !== state.value ) {
+					nextState.value = payload.value;
+					nextState.isDirty = false;
+				}
+				break;
+
+			/**
+			 * Validation
+			 */
+			case actionTypes.INVALIDATE:
+				nextState.error = payload.error;
+				break;
+		}
+
+		if ( payload.event ) {
+			nextState._event = payload.event;
+		}
+
+		/**
+		 * Send the nextState + action to the combinedReducers via
+		 * this "bridge" mechanism. This allows external stateReducers
+		 * to hook into actions, and modify state if needed.
+		 */
+		return combinedStateReducers( nextState, action );
+	};
+}
+
+/**
+ * A custom hook that connects and external stateReducer with an internal
+ * reducer. This hook manages the internal state of InputControl.
+ * However, by connecting an external stateReducer function, other
+ * components can react to actions as well as modify state before it is
+ * applied.
+ *
+ * This technique uses the "stateReducer" design pattern:
+ * https://kentcdodds.com/blog/the-state-reducer-pattern/
+ *
+ * @param {Function} stateReducer An external state reducer.
+ * @param {Object} initialState The initial state for the reducer.
+ * @return {Object} State, dispatch, and a collection of actions.
+ */
+export function useInputControlStateReducer(
+	stateReducer = initialStateReducer,
+	initialState = initialInputControlState
+) {
+	const combinedReducers = composeStateReducers( stateReducer );
```

Comment:
Seems like this is no longer needed, there's nothing to compose.

Reply:
Ah! Good catch 🤗 . Will push a fix for this